x-pack/test/functional/services/ml/test_resources.ts

```diff
@@ -187,44 +197,180 @@ export function MachineLearningTestResourcesProvider({ getService }: FtrProvider
       await this.createSavedSearchIfNeeded(savedSearches.farequoteFilterAndKuery);
     },
 
-    async deleteIndexPattern(title: string) {
-      log.debug(`Deleting index pattern with title '${title}'...`);
+    async deleteSavedObjectById(id: string, objectType: SavedObjectType) {
+      log.debug(`Deleting ${objectType} with id '${id}'...`);
 
-      const indexPatternId = await this.getIndexPatternId(title);
-      if (indexPatternId === undefined) {
-        log.debug(`Index pattern with title '${title}' does not exists. Nothing to delete.`);
+      if ((await this.savedObjectExistsById(id, objectType)) === false) {
+        log.debug(`${objectType} with id '${id}' does not exists. Nothing to delete.`);
         return;
       } else {
         await supertest
-          .delete(`/api/saved_objects/${SavedObjectType.INDEX_PATTERN}/${indexPatternId}`)
+          .delete(`/api/saved_objects/${objectType}/${id}`)
           .set(COMMON_REQUEST_HEADERS)
           .expect(200);
 
-        log.debug(` > Deleted index pattern with id '${indexPatternId}'`);
+        await this.assertSavedObjectNotExistsById(id, objectType);
+
+        log.debug(` > Deleted ${objectType} with id '${id}'`);
+      }
+    },
+
+    async deleteIndexPatternByTitle(title: string) {
+      log.debug(`Deleting index pattern with title '${title}'...`);
+
+      const indexPatternId = await this.getIndexPatternId(title);
+      if (indexPatternId === undefined) {
+        log.debug(`Index pattern with title '${title}' does not exists. Nothing to delete.`);
+        return;
+      } else {
+        await this.deleteIndexPatternById(indexPatternId);
       }
     },
 
-    async deleteSavedSearch(title: string) {
+    async deleteIndexPatternById(id: string) {
+      await this.deleteSavedObjectById(id, SavedObjectType.INDEX_PATTERN);
+    },
+
+    async deleteSavedSearchByTitle(title: string) {
       log.debug(`Deleting saved search with title '${title}'...`);
 
       const savedSearchId = await this.getSavedSearchId(title);
       if (savedSearchId === undefined) {
         log.debug(`Saved search with title '${title}' does not exists. Nothing to delete.`);
         return;
       } else {
-        await supertest
-          .delete(`/api/saved_objects/${SavedObjectType.SEARCH}/${savedSearchId}`)
-          .set(COMMON_REQUEST_HEADERS)
-          .expect(200);
+        await this.deleteSavedSearchById(savedSearchId);
+      }
+    },
+
+    async deleteSavedSearchById(id: string) {
+      await this.deleteSavedObjectById(id, SavedObjectType.SEARCH);
+    },
 
-        log.debug(` > Deleted saved searchwith id '${savedSearchId}'`);
+    async deleteVisualizationByTitle(title: string) {
+      log.debug(`Deleting visualization with title '${title}'...`);
+
+      const visualizationId = await this.getVisualizatzionId(title);
```

Comment:
Typo - should change to `getVisualizationId`

Reply:
Fixed in 64605beecb